test/common/common/inline_map_test.cc

```diff
@@ -0,0 +1,326 @@
+  }
+
+  // Destruct the map.
+  map.reset();
+}
+
+TEST(InlineMapWith20InlineKey, InlineMapWith20InlineKeyTestWithUniquePtrAsValue) {
+  InlineMapDescriptor<std::string> descriptor;
+
+  std::vector<InlineMapDescriptor<std::string>::Handle> handles;
+
+  // Create 20 inline keys.
+  for (size_t i = 0; i < 20; ++i) {
+    handles.push_back(descriptor.addInlineKey("key_" + std::to_string(i)));
+  }
+
+  // Add repeated inline keys will make no effect and return the same handle.
+  for (size_t i = 0; i < 20; ++i) {
+    EXPECT_EQ(handles[i], descriptor.addInlineKey("key_" + std::to_string(i)));
+  }
+
+  descriptor.finalize();
+
+  auto map = InlineMap<std::string, std::unique_ptr<std::string>>::create(descriptor);
+
+  // Set inline entries.
+  for (size_t i = 0; i < 20; ++i) {
+    map->set(handles[i], std::make_unique<std::string>("value_" + std::to_string(i)));
+  }
+
+  // Set dynamic entries.
+  for (size_t i = 20; i < 100; ++i) {
+    map->set("key_" + std::to_string(i),
+             std::make_unique<std::string>("value_" + std::to_string(i)));
+  }
+
+  // Erase entries by typed inline handle.
+  for (size_t i = 0; i < 5; ++i) {
+    map->erase(handles[i]);
+  }
+
+  // Overwrite entries by typed inline handle.
+  for (size_t i = 5; i < 10; ++i) {
+    (*map)[handles[i]] = std::make_unique<std::string>("value_" + std::to_string(i) + "_new");
+  }
+
+  // Erase entries by dynamic key.
+  for (size_t i = 20; i < 25; ++i) {
+    map->erase("key_" + std::to_string(i));
+  }
+
+  // Overwrite entries by dynamic key.
+  for (size_t i = 25; i < 30; ++i) {
+    (*map)["key_" + std::to_string(i)] =
+        std::make_unique<std::string>("value_" + std::to_string(i) + "_new");
+  }
+
+  // Clear the map.
+  map->clear();
+
+  EXPECT_EQ(map->size(), 0);
+
+  // Reset the entries.
+  for (size_t i = 0; i < 100; ++i) {
+    map->set("key_" + std::to_string(i),
+             std::make_unique<std::string>("value_" + std::to_string(i)));
+  }
+
+  EXPECT_EQ(map->size(), 100);
+
+  // Destruct the map.
+  map.reset();
+}
+
+TEST(InlineMapWith20InlineKey, MapCreationWillFinalizeTheDescriptor) {
+  InlineMapDescriptor<std::string> descriptor;
+
+  std::vector<InlineMapDescriptor<std::string>::Handle> handles;
+
+  // Create 20 inline keys.
+  for (size_t i = 0; i < 20; ++i) {
+    handles.push_back(descriptor.addInlineKey("key_" + std::to_string(i)));
+  }
+
+  // Add repeated inline keys will make no effect and return the same handle.
+  for (size_t i = 0; i < 20; ++i) {
+    EXPECT_EQ(handles[i], descriptor.addInlineKey("key_" + std::to_string(i)));
+  }
+
+  auto map = InlineMap<std::string, std::unique_ptr<std::string>>::create(descriptor);
+
+  EXPECT_TRUE(descriptor.finalized());
+}
+
+TEST(InlineMapWith3InlineKey, TestInlineKeysAsString) {
+  InlineMapDescriptor<std::string> descriptor;
+  // Create 3 inline keys.
+  for (size_t i = 0; i < 3; ++i) {
+    descriptor.addInlineKey("key_" + std::to_string(i));
```

Comment:
FYI `absl::StrCat("key_', i)` is faster, as it does't require building a string for each number before concatenating it.

That doesn't matter for this test code of course.

Reply:
Let me do this update next time when I need to add some more tests (I think I will do it soon). I can update all simlilar code in the inline map test at that time. 😸